### PR TITLE
React Context

### PR DESCRIPTION
**Pull request by htjahyon**
image

### CODE REVIEW

File: react-context/client/components/navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                  </a>
+                  <a href="#sign-up" className="btn btn-dark">
+                    Sign Up
+                  </a>
+                </>
+              }
+            </div>
+          </div>
+        </nav>
+      </AppContext.Consumer>
```
Comment (scbowler) [inline react-context/client/components/navbar.jsx]: This isn't how we want to add context to the component, line 5 needs access to the context so using it as JSX won't work for what we need.

File: react-context/client/components/navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    );
+  }
+}
```
Comment (scbowler) [inline react-context/client/components/navbar.jsx:35]: After the class is defined we want to add the context here after the component definition.

```JSX
Navbar.contextType = AppContext;
```

Adding a static property after the component definition will make context available to the entire component which is what we want.

File: react-context/client/pages/auth.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <AuthForm
+              key={route.path}
+              action={route.path}
+              onSignIn={handleSignIn} />
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
```
Comment (scbowler) [inline react-context/client/pages/auth.jsx:36]: This component needs context added to it, in the same way that was discussed for Navbar.

File: react-context/client/pages/home.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        <iframe
+          src="https://giphy.com/embed/Ju7l5y9osyymQ"
+          width="100%"
+          height="100%"
+          frameBorder="0"
+          style={styles.gif} />
+      </div>
+    );
+  }
+}
```
Comment (scbowler) [inline react-context/client/pages/home.jsx:33]: Home needs context added to it

**Review by scbowler — CHANGES_REQUESTED** [review]
Some updates needed, see below comments for details. Let me know if you have any questions.

**Comment by htjahyon** [comment]
image

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback! Looks good! 👍

**Pull request merged**